html/semantics/sections/the-body-element/sections-body-element-mirrors-window.html

```diff
@@ -0,0 +1,178 @@
+//   confirming that "a number of" means all eighteen.
+
+function makeRegEx(strAttr, strArgs, strCode) {
+    // Spec does not specify if function needs to be named or not, so this test accomodates both
+    var regEx = /abc/,
+        strFuncName = "(anonymous|" + strAttr + ")";
+
+    regEx = new RegExp("^function\\s*" + strFuncName + "\\s*\\(" + strArgs + "\\)\\s*\\{\\s*" + strCode + "?\\s*\\}\\s*$", "m");
+
+    return regEx;
+}
+
+// note:  index of attributes here must match return values of actual markup placeholder functions
+var testAttributes = ["onblur", "onfocus", "onload", "onscroll", "onerror", "onafterprint",
+                      "onbeforeprint", "onbeforeunload", "onhashchange", "onmessage", "onoffline",
+                      "ononline", "onpagehide", "onpageshow", "onpopstate", "onresize", "onstorage", "onunload"];
+
+var numAttr = 0,
+    strAttr = "",
+    strCode = "",
+    strArgs = "",
+    strArgsRegEx = "",
+    regEx = /abc/,
+    funcExpected = {},
+    strCheck = "",
+    foobar = "";
+
+
+for (numAttr = 0; numAttr < testAttributes.length; numAttr++) {
+    strAttr = testAttributes[numAttr]; // name of current attribute to check
+    strCode = "foobar = " + numAttr + ";"; // expected content of content attribute
+    // prepare strings needed to create correct event handler (onerror has more arguments than the others)
+    if (strAttr !== "onerror") {
+        strArgs = strArgsRegEx = "event";
+    } else {
+        // is onerror.  Spec defines exactly four parameters for onerror event, so capturing that in this test.
+        strArgs = "event, source, lineno, column";
+        strArgsRegEx = "((event|source|lineno|column),?\\s*){4}"; // for regExp
+    }
+
+    // check that content attribute via markup was correctly read into attributes array
+    test(function () {
+        assert_equals(document.body.attributes[strAttr].value, strCode, "Content attribute " + strAttr + ":");
+    }, strAttr + ": document.body.attributes[" + strAttr + "] populated with content markup for " + strAttr + " attribute");
+
+    // check that body IDL attribute (document.body[strAttr]) mirrors content attribute
+    test(function () {
+        // (As the functions being compared are not the same object, even though the two objects should have the same code,
+        //  we use assert_regexp_match to compare two different function bodies to make sure they are actually the same.)
+
+        funcExpected = new Function(strArgs, strCode);
+        regEx = makeRegEx(strAttr, strArgsRegEx, strCode);
+
+        if ((typeof document.body[strAttr] === "function") && (document.body[strAttr] !== undefined)
+            && (strCheck = document.body[strAttr].toString(), regEx.test(strCheck))) {
+
+        assert_regexp_match(strCheck, regEx, "document.body." + strAttr + ":");
+
+        } else {
+        assert_equals(document.body[strAttr], funcExpected, "document.body." + strAttr + ":");
```

Comment:
This and all other `assert_equals` involving `funcExpected` are broken because it assumes the function object survives the mirroring, but it does not. The spec for this is https://html.spec.whatwg.org/#window-reflecting-body-element-event-handler-set and from that I found existing tests:
https://wpt.fyi/results/html/webappapis/scripting/events/event-handler-attributes-body-window.html?label=experimental
https://wpt.fyi/results/html/webappapis/scripting/events/event-handler-attributes-window-body.html?label=experimental
https://wpt.fyi/results/html/webappapis/scripting/events/event-handler-attributes-windowless-body.html?label=experimental

I'll delete this test from the PR and keep reviewing.